lib/bank_accounts/savings_account.rb

```diff
@@ -0,0 +1,81 @@
+module BankAccounts
+  class SavingsAccount < Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+    # limits
+    MINIMUM_BALANCE = 10
+
+    # fees
+    WITHDRAWAL_FEE = 2.00
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    # create a new savings account.
+    def initialize(id, initial_balance)
+      if initial_balance < MINIMUM_BALANCE
+        raise ArgumentError.new("You cannot create a savings account with less than a $#{ MINIMUM_BALANCE } initial deposit.")
+      end
+
+      super(id, initial_balance)
+
+      return true
+    end
+
+
+    # add interest to account.
+    def add_interest(rate_percentage)
+      if validate_number(rate_percentage)
+        # convert rate from percentage to decimal.
+        rate_decimal = rate_percentage / 100.0
+
+        # calculate interest.
+        interest = @balance * rate_decimal
+
+        # add interest to balance.
+        update_balance(interest)
+
+        # return interest.
+        return interest
+      else
+        return false
+      end
+    end
+
+
+    # withdraw money from account.
+    def withdraw(amount)
+      if super(amount + WITHDRAWAL_FEE)
+        return @balance
+      else
+        return @balance
+      end
+    end
```

Comment:
I was just rereading my updates to my code before merging the pull request, and I realized this comment still confuses me a bit. Can you give me some advice on when to use implicit returns? For my way of thinking, it usually makes more sense to spell out returns explicitly so I don't have to go back and research them later. Is it better to use implicit in this case -- when it involves checking out the code this code inherits from to make sure it's still returning the balance per the specifications?